crates/env/src/call/call_builder.rs

```diff
@@ -109,7 +109,28 @@ where
     /// Invokes the contract with the given built-up call parameters.
     ///
     /// Returns the result of the contract execution.
+    ///
+    /// # Panics
+    ///
+    /// This method panics if it encounters an [`ink_primitives::LangError`]. If you want to handle
+    /// those use the [`try_invoke`][`CallParams::try_invoke`] method instead.
     pub fn invoke(&self) -> Result<R, crate::Error> {
+        crate::invoke_contract(self).map(|inner| {
+            inner.expect(
+                "Handling `LangError`s is not supported by `CallParams::invoke`, use \
+                `CallParams::try_invoke` instead.",
+            )
```

Comment:
This drops the `LangError`. I think we should just `unwrap` or compose a meaningful error message that includes the error: "Call failed with error {}"